File: inst/htmlwidgets/utils/history_api.js
//**
// * History API
// * Functions implementing interactions with the history of a tabulator table.
// * See https://tabulator.info/docs/6.0/history
// */


const undoAction = params => {
  const tableId = params.table_id;
  table = getTabulatorTable(tableId);
  table.undo();
};

const redoAction = params => {
  const tableId = params.table_id;
  table = getTabulatorTable(tableId);
  table.redo();
};

const clearHistory = params => {
  const tableId = params.table_id;
  table = getTabulatorTable(tableId);
  table.clearHistory();
};

Shiny.addCustomMessageHandler("undo_action", undoAction);
Shiny.addCustomMessageHandler("redo_action", redoAction);
Shiny.addCustomMessageHandler("clear_history", clearHistory)

File: R/history_api.R
#' Undo an action for the given table
#' @param table_id The ID of the table
#' @param session The Shiny session object
#' @importFrom shiny getDefaultReactiveDomain
#' @export
undo_action <- function(table_id, session = shiny::getDefaultReactiveDomain()) {
  session$sendCustomMessage(
    type = "undo_action",
    message = list(table_id = get_namespaced_id(table_id, session))
  )
}

#' Redo an action for the given table
#' @param table_id The ID of the table
#' @param session The Shiny session object
#' @importFrom shiny getDefaultReactiveDomain
#' @export
redo_action <- function(table_id, session = shiny::getDefaultReactiveDomain()) {
  session$sendCustomMessage(
    type = "redo_action",
    message = list(table_id = get_namespaced_id(table_id, session))
  )
}

#' Clear history for the given table
#' @param table_id The ID of the table
#' @param session The Shiny session object
#' @importFrom shiny getDefaultReactiveDomain
#' @export
clear_history <- function(table_id, session = shiny::getDefaultReactiveDomain()) {
  session$sendCustomMessage(
    type = "clear_history",
    message = list(table_id = get_namespaced_id(table_id, session))
  )
}
